api/utils/keys/privatekey.go

```diff
@@ -0,0 +1,143 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package keys defines common interfaces for Teleport client keys.
+package keys
+
+import (
+	"crypto"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"os"
+
+	"github.com/gravitational/trace"
+	"golang.org/x/crypto/ssh"
+	"golang.org/x/crypto/ssh/agent"
+)
+
+const (
+	rsaPrivateKeyType = "RSA PRIVATE KEY"
+)
+
+// PrivateKey implements crypto.PrivateKey and crypto.Signer, with additional helper methods
```

Comment:
If this interface implements `crypto.PrivateKey` and `crypto.Signer` then why on `crypto.Signer` is embedded?

Reply:
`crypto.PrivateKey` is just an `any` type, and has to be type asserted to be used. Here's the godoc for `crypto.PrivateKey`:

```
PrivateKey represents a private key using an unspecified algorithm.

Although this type is an empty interface for backwards compatibility reasons, all private key types in the standard library implement the following interface

interface{
    Public() crypto.PublicKey
    Equal(x crypto.PrivateKey) bool
}
as well as purpose-specific interfaces such as Signer and Decrypter, which can be used for increased type safety within applications.
```